weave-js/src/common/components/WBMenu.tsx

```diff
@@ -106,130 +100,69 @@ export const WBMenu = React.forwardRef<HTMLDivElement, WBMenuProps>(
       fontSize,
       lineHeight,
       highlightFirst,
-      highlighted: highlightedProp,
-      onChangeHighlighted,
       onSelect,
       onEsc,
       dataTest,
     },
     ref
   ) => {
-    const getDefaultHoverIndex = React.useCallback(() => {
-      let found = -1;
-      if (highlightFirst && options.length > 0) {
-        found = options.findIndex(opt => !opt.disabled);
-      } else {
-        found = options.findIndex(opt => selected === opt.value);
-      }
+    const [defaultHighlighted, defaultHighlightedIndex] = useMemo(() => {
+      if (!highlightFirst) return [undefined, undefined];
+      const foundIndex = options.findIndex(el => !el.disabled);
 
-      return found >= 0 ? found : 0;
-    }, [highlightFirst, options, selected]);
-    // dedupe options
-    const prevLength = options.length;
-    options = _.uniqBy(options, opt => opt.value);
-    if (options.length !== prevLength) {
-      // eslint-disable-next-line no-console
-      console.warn('Passed duplicate options into WBMenu');
-    }
-
-    const [highlighted, setHighlighted] = useControllableState(
-      options[getDefaultHoverIndex()]?.value || null,
-      highlightedProp,
-      onChangeHighlighted
-    );
-    const highlightedIndex = useMemo(() => {
-      if (highlighted === null) {
-        return null;
-      }
-      const found = options.findIndex(opt => highlighted === opt.value);
+      if (!foundIndex) return [undefined, undefined];
```

Comment:
do we use single line returns? I feel like i never see them

Reply:
Not sure, I'm just so used to them in C++ that it is force of habit... I can switch out for the more verbose option though